devito/passes/iet/definitions.py

```diff
@@ -409,12 +412,18 @@ def place_casts(self, iet, **kwargs):
 
         return iet, {}
 
+    @iet_pass
+    def make_langtypes(self, iet):
+        iet, metadata = lower_complex(iet, self.lang, self.compiler)
+        return iet, metadata
+
     def process(self, graph):
         """
         Apply the `place_definitions` and `place_casts` passes.
         """
         self.place_definitions(graph, globs=set())
         self.place_casts(graph)
+        self.make_langtypes(graph)
```

Comment:
I think you're calling this in the wrong place.

It should be called within https://github.com/devitocodes/devito/blob/master/devito/operator/operator.py#L480

that is potentially somewhere here:

https://github.com/devitocodes/devito/blob/master/devito/passes/iet/misc.py#L166

Note the name `generate_macros` is somewhat legacy, but you get why I'm suggesting to move it in there